clang/test/SemaOpenCL/invalid-kernel-attrs.cl

```diff
@@ -10,7 +10,9 @@ kernel __attribute__((vec_type_hint(bool))) void kernel4() {} //expected-error{{
 
 kernel __attribute__((vec_type_hint(int))) __attribute__((vec_type_hint(float))) void kernel5() {} //expected-warning{{attribute 'vec_type_hint' is already applied with different arguments}}
 
-kernel __attribute__((work_group_size_hint(8,16,32,4))) void kernel6() {} //expected-error{{'work_group_size_hint' attribute requires exactly 3 arguments}}
+kernel __attribute__((work_group_size_hint(8,16,32,4))) void kernel6() {} //expected-error{{'work_group_size_hint' attribute takes no more than 3 arguments}}
```

Comment:
@AaronBallman allowing optional values in the attribute changes this the wording of this error message, is it ok?

Reply:
Yes, so in `handleWorkGroupSizeHint()` I reject the case "GNU spelling + less than three args", but the diagnostic for the attribute spelling is already handled "for us" (and thus the wording change as we move from exactly three to max 3 with the missing optionals). I don't really see a nice solution to that

Reply:
Ahhhh, thanks, that was the bit I was not thinking about! There is common attribute checking that happens here: https://github.com/intel/llvm/blob/sycl/clang/lib/Sema/SemaDeclAttr.cpp#L10269.

As a temporary workaround, you can specify `HasCustomParsing = 1` in Attr.td for the attribute (that will skip all the argument checking but retain the other checking) and then add that checking back in manually in the attribute handler, but it might be nice for us to make some changes here: https://github.com/intel/llvm/blob/sycl/clang/lib/Sema/SemaDeclAttr.cpp#L10206 so that we can skip argument counting for more than just attributes accepting a pack. I'm not certain what the best approach is there yet, but the temporary workaround should suffice for now hopefully.